### PR TITLE
Added exclude parameter to API

### PR DESCRIPTION
**Pull request by kaitlin**
- Added `exclude` param to API
- Added a unique `id` field to the api response/serializer 
- Updated README to document usage of `exclude` param
- Added a test
- Updated the `assertResultsEqual` method to test only for keys in the expected result. This will avoid needing to update every test when an attribute is added to the api response.

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -65,9 +66,16 @@ def get_contracts_queryset(request_params, wage_field):
     sort = request_params.get('sort', wage_field)
     # query_type can be: [ match_all (default) | match_phrase | match_exact ]
     query_type = request_params.get('query_type', 'match_all')
+    exclude = request_params.getlist('exclude')
+    if len(exclude) == 1 and ',' in exclude[0]:
```
Comment (arowla) [inline api/views.py]: Do you even need to do this check? `.split(',')` will return a single-valued list if there is no match for the separator.

File: api/tests.py [hunk 2]
```diff
@@ -598,6 +598,22 @@ def test_filter_by_site(self):
            'contractor_site': 'Q3',
            'business_size': None}])
 
+    def test_exclude_by_id(self):
+        get_contract_recipe().make(_quantity=1, id=100)
```
Comment (arowla) [inline api/tests.py]: I _think_ you should be able to do `get_contract_recipe().make(_quantity=3, id=cycle([100,101,102]))`

File: api/views.py (hunk 1 above)
Comment (kaitlin) [inline api/views.py]: Good point. I will remove.

File: api/tests.py (hunk 2 above)
Comment (kaitlin) [inline api/tests.py]: This is what was throwing the error. It tries to cast a generator object to an int and fails. The error is `TypeError: int() argument must be a string or a number, not 'itertools.cycle'`. I feel like this is a limitation of mommy model so I enumerated the calls as a work around.

Comment (arowla) [inline api/tests.py]: Hm. Maybe primary keys are special. Perhaps you should file a ticket over at model_mommy.

Do you still need the `_quantity=1` param, in that case?

Comment (kaitlin) [inline api/tests.py]: Derp, no I don't. Removed.

**Pull request merged**